bin/bigquery.py

```diff
@@ -14,158 +14,202 @@
+            Argument(
+                name="service_account",
+                title="Service Account",
+                description="GCP Service Account as a single line of JSON",
+                data_type=Argument.data_type_string,
+                required_on_create=True,
+            )
+        )
+        scheme.add_argument(
+            Argument(
+                name="time_field",
+                title="Time Field",
+                description="The column that contains the event time",
+                data_type=Argument.data_type_string,
+                required_on_create=False,
+            )
+        )
+        scheme.add_argument(
+            Argument(
+                name="checkpoint_field",
+                title="Checkpoint Field",
+                description="The column that increases in newer events, last value will replace %checkpoint% in query.",
+                data_type=Argument.data_type_string,
+                required_on_create=False,
+            )
+        )
+        scheme.add_argument(
+            Argument(
+                name="checkpoint_start_value",
+                title="Checkpoint start value",
+                description="Defaults to 0 if not set. If you want to start sync from a certain point or use a TIMESTAMP type field you can set a different value here, for example 1970-01-01 00:00:00 UTC",
+                data_type=Argument.data_type_string,
+                required_on_create=False,
+            )
+        )
+        scheme.add_argument(
+            Argument(
+                name="blacklist",
+                title="Field Blacklist",
+                description="Comma seperated list of columns that should not be indexed (such as the checkpoint field)",
+                data_type=Argument.data_type_string,
+                required_on_create=False,
+            )
+        )
         return scheme
 
     def stream_events(self, inputs, ew):
-        self.service.namespace['app'] = self.APP
-        
+        self.service.namespace["app"] = self.APP
+
         # Get Variables
         input_name, input_items = inputs.inputs.popitem()
         kind, name = input_name.split("://")
         checkpointfile = os.path.join(
             self._input_definition.metadata["checkpoint_dir"],
-            "".join([c for c in name if c.isalpha() or c.isdigit() or c==' ']).rstrip()
+            "".join(
+                [c for c in name if c.isalpha() or c.isdigit() or c == " "]
+            ).rstrip(),
         )
 
         query = input_items["query"]
-        time_field = input_items.get('time_field')
-        checkpoint_field = input_items.get('checkpoint_field')
-        blacklist = input_items.get('blacklist','').split(',')
-        
+        time_field = input_items.get("time_field")
+        checkpoint_field = input_items.get("checkpoint_field")
+        checkpoint_start_value = input_items.get("checkpoint_start_value") or "0"
+        blacklist = (input_items.get("blacklist") or "").split(",")
```

Comment:
Changed this as I got bitten by first setting this field then clearing it and then getting `'NoneType' object has no attribute 'split'` errors